Stateless/StateRepresentation.cs

```diff
@@ -11,7 +11,7 @@ internal class StateRepresentation
         {
             readonly TState _state;
 
-            readonly IDictionary<TTrigger, ICollection<TriggerBehaviour>> _triggerBehaviours =
+            protected internal readonly IDictionary<TTrigger, ICollection<TriggerBehaviour>> _triggerBehaviours =

```

Comment:
Rather than making the fields `protected internal` could we wrap them in `internal` properties?